File: java-pkmst/src/main/java/com/prokarma/pkmst/model/RequestTimeFilterProperty.java
package com.prokarma.pkmst.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonValue;
/**
 * Response class to be returned by Api
 * @author pkmst
 *
 */
import com.fasterxml.jackson.annotation.JsonCreator;

/**
 * Gets or Sets RequestTimeFilterProperty
 */
public enum RequestTimeFilterProperty {
  
  TRAVEL_TIME("travel_time"),
  
  DISTANCE("distance"),
  
  DISTANCE_BREAKDOWN("distance_breakdown"),
  
  FARES("fares"),
  
  ROUTE("route");

  private String value;

  RequestTimeFilterProperty(String value) {
    this.value = value;
  }

  @Override
  @JsonValue
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static RequestTimeFilterProperty fromValue(String text) {
    for (RequestTimeFilterProperty b : RequestTimeFilterProperty.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + text + "'");
  }
}

File: kotlin-spring/src/main/kotlin/org/openapitools/model/RequestTimeFilterProperty.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid

/**
* 
* Values: travelTime,distance,distanceBreakdown,fares,route
*/
enum class RequestTimeFilterProperty(val value: kotlin.String) {

    @JsonProperty("travel_time") travelTime("travel_time"),

    @JsonProperty("distance") distance("distance"),

    @JsonProperty("distance_breakdown") distanceBreakdown("distance_breakdown"),

    @JsonProperty("fares") fares("fares"),

    @JsonProperty("route") route("route");

}

